sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/batching/IndexingDocumentManager.java

```diff
@@ -26,13 +26,10 @@
  * @param <T> The type of document that is being indexed.
  */
 final class IndexingDocumentManager<T> {
-    private final ClientLogger logger;
-
     private final LinkedList<TryTrackingIndexAction<T>> actions = new LinkedList<>();
-    private final Semaphore semaphore = new Semaphore(1);
+    private final ReentrantLock lock = new ReentrantLock();
```

Comment:
What were the differences you saw between `Semaphore` and `ReentrantLock` that made you swap them out?

Reply:
Semaphores can be acquired and released by different threads, where ReentrantLock can only be released by the locking thread. Since these cases are synchronous, ReentrantLock is a better choice as only one thread should be in this code path at any given time.